### PR TITLE
[PIPELINES] m lock flag for vault to secure settings and removed external vault from readme until pipelines-1.9.0 supports it.

### PR DESCRIPTION
**Pull request by peters95**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by jefferyfry — APPROVED** [review]
LGTM

**Comment by peters95** [comment]
Need to add the security context to the deployment yaml of the vault if mlock is enabled 

     securityContext:
       capabilities:
         add:
           - IPC_LOCK

File: stable/pipelines/README.md [hunk 1]
```diff
@@ -216,6 +187,18 @@ To start using Pipelines you need to setup a Build Plane:
...
+
+https://www.vaultproject.io/docs/configuration#disable_mlock
+
+For dev or test environment it is acceptable to leave this value set to true.
+
+Note however this does enable a potential security issue where encrypted credentials could potentially be swapped onto an unencrypted disk. 
+
+For this reason we recommend you always set this value to false to ensure mlock is enabled.
+
+
```
Comment (rimusz) [inline stable/pipelines/README.md:201]: Please add an example from the values.yaml how to set to `true` or false`

**Review by rimusz — CHANGES_REQUESTED** [review]

File: stable/pipelines/values.yaml [hunk 2]
```diff
@@ -1047,6 +1046,12 @@ vault:
     type: ClusterIP
     port: 30100
 
+  # https://www.vaultproject.io/docs/configuration#disable_mlock
+  # disable_mlock (bool: false) – Disables the server from executing the mlock syscall.
+  # mlock prevents memory from being swapped to disk.
+  # Disabling mlock is not recommended in production, but is fine for local development and testing.
+  disablemlock: true
```
Comment (rimusz) [inline stable/pipelines/values.yaml]: this one should be set then to `false` by default

**Review by rimusz — CHANGES_REQUESTED** [review]

File: stable/pipelines/values.yaml [hunk 3]
```diff
@@ -1047,10 +1047,10 @@ vault:
     port: 30100
 
   # https://www.vaultproject.io/docs/configuration#disable_mlock
-  # disable_mlock (bool: false) – Disables the server from executing the mlock syscall.
-  # mlock prevents memory from being swapped to disk.
   # Disabling mlock is not recommended in production, but is fine for local development and testing.
-  disablemlock: true
+  # true - dev or test mlock is diabled
+  # false - production mlock is enabled
+  disablemlock: false
```
Comment (rimusz) [inline stable/pipelines/values.yaml:1053]: please change back to `true`

**Review by rimusz — CHANGES_REQUESTED** [review]

File: stable/pipelines/values.yaml (hunk 3 above)
Comment (peters95) [reply stable/pipelines/values.yaml]: Updated to true

File: stable/pipelines/README.md (hunk 1 above)
Comment (peters95) [reply stable/pipelines/README.md:201]: Added to values.yaml that you can set it to true for dev or test and false for production grade systems.

**Review by rimusz — APPROVED** [review]

**Review by vinayagg — APPROVED** [review]

**Pull request merged**